core/src/main/scala/cromwell/core/labels/Label.scala

```diff
@@ -5,26 +5,37 @@ import cats.data.Validated._
 import cats.syntax.cartesian._
 import cats.syntax.validated._
 
+import scala.util.matching.Regex
+
 sealed abstract case class Label(key: String, value: String)
 
 object Label {
 
   // Yes, 63. Not a typo for 64.
   // See 'labels' in https://cloud.google.com/genomics/reference/rpc/google.genomics.v1alpha2#google.genomics.v1alpha2.RunPipelineArgs
   private val MaxLabelLength = 63
-  val LabelRegexPattern = "[a-z]([-a-z0-9]*[a-z0-9])?"
+  val GoogleLabelRegexPattern = "[a-z]([-a-z0-9]*[a-z0-9])?"
+  val LabelKeyRegex = GoogleLabelRegexPattern
+  val LabelValueRegex = "([a-z0-9]*[-a-z0-9]*[a-z0-9])?"
+
+  val LabelExpectationsMessage =
+    s"A Label key must match the pattern `$LabelKeyRegex` and a label value must match the pattern `$LabelValueRegex`."
 
-  def validateName(s: String): ErrorOr[String] = {
-    if (LabelRegexPattern.r.pattern.matcher(s).matches) {
-      if (s.length <= MaxLabelLength) s.validNel else s"Invalid label: $s was ${s.length} characters. The maximum is $MaxLabelLength".invalidNel
+  def validateLabelRegex(s: String, regexAllowed: Regex): ErrorOr[String] = {
+    if (regexAllowed.pattern.matcher(s).matches) {
```

Comment:
pattern match on `(regexAllowed.pattern.matcher(s).matches, s.length <= MaxLabelLength)` could clear up the nested if's.  A little easier to read